modules/core/multihop/proof_gen.go

```diff
@@ -26,10 +26,13 @@ type Endpoint interface {
 	GetConnection() (*connectiontypes.ConnectionEnd, error)
 	// Returns the proof of the `key`` at `height` within the ibc module store.
 	QueryProofAtHeight(key []byte, height int64) ([]byte, clienttypes.Height, error)
+	QueryStateAtHeight(key []byte, height int64) []byte
+	QueryMinimumConsensusHeight(minHeight exported.Height, maxHeight exported.Height) (exported.Height, exported.Height, error)
 	GetMerklePath(path string) (commitmenttypes.MerklePath, error)
 	// UpdateClient updates the clientState of counterparty chain's header
 	UpdateClient() error
 	Counterparty() Endpoint
+	Debug()
```

Comment:
Should we remove this method?

Reply:
Is `UpdateClient()` method still required considering that Relayers don't need to aggressively update all clients in the channel path?

Reply:
It is no longer needed removing it plus underlying methods which are not required!